File: fix_versions.py
import os
import re

def remove_version_from_imports(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # @숫자.숫자.숫자 패턴을 찾아서 제거
    new_content = re.sub(r'"([^"]*@)\d+\.\d+\.\d+"', r'"\1"', content)

    if content != new_content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {file_path}")
        return True
    return False

def process_directory(dir_path):
    updated_count = 0
    for root, dirs, files in os.walk(dir_path):
        for file in files:
            if file.endswith('.tsx'):
                file_path = os.path.join(root, file)
                if remove_version_from_imports(file_path):
                    updated_count += 1
    return updated_count

updated = process_directory('nextjs-training/components/ui')
print(f"Total files updated: {updated}")

File: fix_versions.js
const fs = require('fs');
const path = require('path');

// 버전 패턴을 제거하는 함수
function removeVersionFromImport(line) {
  // @숫자.숫자.숫자 패턴을 찾아서 제거
  return line.replace(/"([^"]*@)\d+\.\d+\.\d+"/g, '"$1"');
}

// components/ui 폴더 내 모든 .tsx 파일 처리
function processDirectory(dirPath) {
  const files = fs.readdirSync(dirPath);

  files.forEach(file => {
    const filePath = path.join(dirPath, file);
    const stat = fs.statSync(filePath);

    if (stat.isDirectory()) {
      processDirectory(filePath);
    } else if (file.endsWith('.tsx')) {
      console.log(`Processing ${filePath}`);

      const content = fs.readFileSync(filePath, 'utf8');
      const lines = content.split('\n');
      const processedLines = lines.map(removeVersionFromImport);
      const newContent = processedLines.join('\n');

      if (content !== newContent) {
        fs.writeFileSync(filePath, newContent);
        console.log(`Updated ${filePath}`);
      }
    }
  });
}

processDirectory('nextjs-training/components/ui');
console.log('All files processed!');
